### DISCUSSION TITLE
What "how‐to guides" do you want to see published on the wiki?

### DISCUSSION THREAD

**Discussion by SlavaVedernikov** [SlavaVedernikov/C4InterFlow#101] [Ideas]
What is it that you are trying to do with C4InterFlow?
Are you not quite sure how to go about it?

Just ask, and we'll write up a guide for that.

**Comment by pietgk** [comment]
Hi,

I am working on extracting C4 dsl files from our repos (nextjs, expo apps, react frontends, aws tRpc backend microservices, ...)
Its very much work in progress and experimental.

your AaC and BPaC with C4 and interface and flow look like a great way to present our architecture and business processes.

When i am a bit further in the process and let C4InterFlow sync in a bit more we can see if a how to for typescript based codebases is an option :-)

If you know of others that are working on the same thing and want to share experiences then feel free to let me know.

Thx Piet

And up front thanks for sharing C4InterFlow with us!

**Reply by SlavaVedernikov** [reply]
Hi Piet,

Yes, that is s good candidate.

I haven't tried extracting AaC from TypeScript codebases automatically, but I think the approach would be very similar to the one I used for C# codebase in [dotnet.eShop sample](https://github.com/SlavaVedernikov/C4InterFlow/tree/master/Samples/dotnet.eShop).

[traversets/code-extractor](https://github.com/olasunkanmi-SE/ts-codebase-analyzer) seems to be doing a good job extracting Abstract Syntax Tree (AST) from  TypeScript and JavaScript codebases.

You'd need to iterate through files, classes, methods, invocations etc.

**Comment by SlavaVedernikov** [comment]
@pietgk Here is a good article about the use of another package - [ts-morph](https://github.com/dsherret/ts-morph) for extracting architecture from TypeScript codebase [Extracting your software architecture with ts-morph](https://medium.com/nick-tune-tech-strategy-blog/extracting-your-software-architecture-with-ts-morph-73059771d2d6)

**Reply by pietgk** [reply]
thx for the link, and yes i am using ts-morph :-)
